packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -475,6 +481,45 @@ async function handler(
     ) {
       throw new HttpError({ statusCode: 403, message: ErrorCode.CancelledBookingsCannotBeRescheduled });
     }
+    const userReschedulingIsOwner = isUserReschedulingOwner(userId, originalRescheduledBooking?.user?.id);
+    let isTeamOwnerOrAdmin = false;
+    if (isTeamEventType) {
+      const teamOwnerOrAdmin = await prisma.membership.findFirst({
+        where: {
+          teamId: eventType?.teamId || 0,
+          userId,
+          role: {
+            in: [MembershipRole.ADMIN, MembershipRole.OWNER],
+          },
+        },
+        select: {
+          userId: true,
+        },
+      });
+      isTeamOwnerOrAdmin = !!teamOwnerOrAdmin;
+    }
+    if (userReschedulingIsOwner || isTeamOwnerOrAdmin) {
+      const attendeesList = originalRescheduledBooking?.attendees.map((attendee) => attendee.email);
+      const users = await prisma.user.findMany({
+        where: {
+          email: {
+            in: attendeesList,
+          },
+        },
+        select: {
+          credentials: {
+            select: credentialForCalendarServiceSelect,
+          }, // Don't leak to client
```

Comment:
NIT: Why was this comment added? I am not sure if it is needed, please remove it.

Reply:
Done